src/main/java/edu/cornell/kfs/concur/batch/service/impl/ConcurSaeCreateRequestedCashAdvanceFileServiceImpl.java

```diff
@@ -0,0 +1,145 @@
+    protected ConcurSaeCreateRequestedCashAdvanceFileValidationService concurSaeCreateRequestedCashAdvanceFileValidationService;
+
+    public List<String> getUnprocessedSaeFiles() {
+        List<String> fileNamesToLoad = getBatchInputFileService().listInputFileNamesWithDoneFile(getBatchInputFileType());
+        return fileNamesToLoad;
+    }
+
+    public boolean processFile(String standardAccountingExtractFullyQualifiedFileName) {
+        boolean processingSuccessful = false;
+        ConcurStandardAccountingExtractFile standardAccountingExtractFile = loadFileIntoParsedDataObject(standardAccountingExtractFullyQualifiedFileName);
+        ConcurSaeRequestedCashAdvanceBatchReportData reportData = new ConcurSaeRequestedCashAdvanceBatchReportData();
+        reportData.setConcurFileName(parseStandardAccountingExtractFileNameFrom(standardAccountingExtractFullyQualifiedFileName));
+        if (getConcurSaeCreateRequestedCashAdvanceFileValidationService().saeHeaderRowValidatesToFileContentsForRequestedCashAdvances(standardAccountingExtractFile, reportData)) {
+            List<String> uniqueRequestedCashAdvanceIdsInFile = new ArrayList<String>();
+            for (ConcurStandardAccountingExtractDetailLine detailFileLine : standardAccountingExtractFile.getConcurStandardAccountingExtractDetailLines()) {
+                getConcurSaeCreateRequestedCashAdvanceFileValidationService().performSaeDetailLineValidationForRequestedCashAdvances(detailFileLine, uniqueRequestedCashAdvanceIdsInFile);
+            }
+            standardAccountingExtractFile.setOriginalFileName(parseStandardAccountingExtractFileNameFrom(standardAccountingExtractFullyQualifiedFileName));
+            processingSuccessful = getConcurCreateCashAdvancePdpFeedFileService().createPdpFeedFileForValidatedDetailFileLines(standardAccountingExtractFile, reportData);
+            if (processingSuccessful && StringUtils.isNotBlank(standardAccountingExtractFile.getFullyQualifiedRequestedCashAdvancesPdpFileName())) {
+                try {
+                    getConcurCreateCashAdvancePdpFeedFileService().createDoneFileForPdpFile(standardAccountingExtractFile.getFullyQualifiedRequestedCashAdvancesPdpFileName());
+                } catch (IOException ioe) {
+                    LOG.error("processFile: ConcurCreateCashAdvancePdpFeedFileService().createDoneFileForPdpFile generated IOException attempting to create .done file for generated PdpFeedFile: " + standardAccountingExtractFile.getFullyQualifiedRequestedCashAdvancesPdpFileName());
+                    processingSuccessful = false;
+                } catch (FileStorageException fse) {
+                    LOG.error("processFile: ConcurCreateCashAdvancePdpFeedFileService().createDoneFileForPdpFile generated FileStorageException attempting to create .done file for generated PdpFeedFile: " + standardAccountingExtractFile.getFullyQualifiedRequestedCashAdvancesPdpFileName());
```

Comment:
another place to include the exception in the log

Reply:
adjusted code to include